### DISCUSSION TITLE
Railgun/Coilguns Harpoons issues

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14643] [Unstable feedback]
Harpoon ammo for turrets was very highly anticipated, but

1. Pull force value is too high
Currently the pull values of the Railgun (and coilgun) harpoons are too big and completely bypass mass or physics.

Examples:
Harpooning a moloch as if it was made of light feathers, which then propels you at extremely speeds
![image](https://github.com/user-attachments/assets/f2c1929e-bc46-4431-b246-c30445d806ac)

Dragging a fully flooded Humpback up with a single railgun harpoon with no slowing down

https://github.com/user-attachments/assets/0b78d025-1ff0-42e8-b963-7f9c30195db0

A single box of coilgun harpoons holding a Charybdis
![image](https://github.com/user-attachments/assets/d3a54119-7ff0-465e-bd9d-baf13f98d621)

Even a single coilgun harpoon will be stronger than a hammerhead (the biggest "small" enemy)

Compare to [Movable and Sellable Wrecks](https://steamcommunity.com/sharedfiles/filedetails/?id=2772926011) where we used a harpoon (grapple) depth charge, so you actually had to drain some of the wreck to pull it.

```xml
movable wrecks grapple
      <Rope targetpullforce="60000" projectilepullforce="0" sourcepullforce="70000" maxlength="10000" snaponcollision="true" spritewidth="20" tile="true" origin="0.05,0.5" targetminmass="0" lerpforces="true">

vanilla railgun harpoon  (over 3x the pull power)
    <Rope targetpullforce="200000" sourcepullforce="10000" projectilepullforce="10" maxlength="10000" snaponcollision="true" spritewidth="10" BarrelLengthMultiplier="0.3" origin="0.05,0.5" tile="true">
````

2. Very little sourcepullforce
There is no force applied to the main sub shooting the harpoon, which means the harpoon drags the target 20x as much as the source, when it should be equal

3. Harpoons don't apply drag/slow affliction
When harpooning monsters, they don't apply a drag/slow affliction to the monster (unlike regular harpoons). 
The drag affliction simulates the monster not being able to swim away without cranking up the pull force to be much stronger than its ability to swim. In other words it keeps the pull force reasonable while it helps dragging bigger monsters

4. Projectiles can't stick to level walls
Source pull force + Harpoon lodged in ceiling = spidersub
You could in theory slow down the fall of the submarine by harpooning the ceiling in case of emergency

5. No decorative sprite
![image](https://github.com/user-attachments/assets/a1baadac-72b6-4817-8c0d-4faab1d0f9cc)

**Comment by Regalis11** [comment]
Fixed in our private repo, fixes coming in the next unstable build.
